ecommerce.js

```diff
@@ -0,0 +1,108 @@
+const express = require('express');
+const mysql = require('mysql2');
+const bodyParser = require('body-parser');
+const { check, validationResult } = require('express-validator');
+
+const connection = mysql.createConnection({
+    host: 'localhost',
+    user: 'sathish',
+    password:'Sathish@19',
+    database: 'ecommerce'
+  });
+
+connection.connect((err) => {
+    if(err){
+        throw err;
+    }
+    console.log('mysql is connected....')
+})
+
+const app = express();
+app.use(express.static('public'));
+
+app.set('view engine', 'ejs');
+
+app.use(bodyParser.urlencoded({ extended: false }));
+
+app.post('/', [
+  check('username','Username must be 3+ char long')
+    .exists()
+    .isLength({ min : 3 }),
+  check('email', 'Invalid Email')
+    .isEmail()
+    .normalizeEmail(),
+ 
+],(req,res) => {
+  const errors = validationResult(req);
+    if(!errors.isEmpty()){
+      const alert = errors.array();
+      res.render('signup',{ alert });
+    }
+  
+  const { email, username, password } = req.body;
+  let check = 'select * from customer';
+  let count = 0;
+  connection.query(check,
+        function(err, results) {
+            if(err){
+                throw err;
+            }
+            for(let i=0;i<results.length;i++){
+              if(results[i].email === email){
+                count++;
+              }
+            }
+            if(count > 0){
+              res.render('validation',{count});
+            }
+            else{
+              let sqlCode = `INSERT INTO customer(username,password,email) VALUES('${username}','${password}','${email}')`;
```

Comment:
This line will create an SQL injection. You should avoid string concatenation in the queries

Reply:
Okay manoj. I will look for the solution and correct it